src/firefly/python/display/FireflyClient.py

```diff
@@ -242,35 +277,53 @@ def uploadTextData(self, stream):
 
 
     def uploadData(self, stream, dataType):
-        url = 'http://' + self.thisHost + '/fftools/sticky/Firefly_FileUpload?preload='
+        url = 'http://' + self.thisHost + '/firefly/sticky/Firefly_FileUpload?preload='
         url+= 'true&type=FITS' if dataType=='FITS' else 'false&type=UNKNOWN'
         dataPack= {'data' : stream}
         result = self.session.post(url, files=dataPack)
         index = result.text.find('$')
         return result.text[index:]
 
 
-    def showFitsOLD(self, fileOnServer=None, plotId=None, additionalParams=None):
-        """ Show a fits image
-        :param: fileOnServer: the is the name of the file on the server.  If you used uploadFile()
-                          then it is the return value of the method. Otherwise it is a file that
-                          firefly has direct read access to.
-        :param: plotId: the id you assigned to the plot. This is necessary to further control the plot
-        :param: additionalParam: dictionary of any valid fits viewer plotting parameters,
-                          see firefly/docs/fits-plotting-parameters.md
-        :return: status of call
+
+    def dispatchRemoteAction(self, channel, actionType, payload):
         """
-        url = self.urlRoot + '?cmd=pushFits'
-        if additionalParams:
-            url+= '&' + '&'.join(['%s=%s' % (k, v) for (k, v) in additionalParams.items()])
-        if plotId:
-            url+= '&plotId=%s' % plotId
-        if fileOnServer:
-            url+= '&file=%s' % fileOnServer
+        dispatch the action to the server by using 'GET' request
+        :param: channel
+        :param: actionType
+        :param: payload: payload for the action
+        """
+
+        action= {'type':actionType, 'payload':payload}
+        url = self.urlRoot + '?channelID=' + channel + '&cmd=pushAction&action='
+        url+= json.dumps(action)
         return self.sendURLAsGet(url)
 
+
+    def dispatchRemoteActionByPost(self, channel, actionType, payload):
+        """
+        dispatch the action to the server by using 'POST' request
+        :param: channel
+        :param: actionType
+        :param: payload: payload for the action
+        """
+        action={'type': actionType, 'payload': payload}
+        data = {'channelID': channel, 'cmd':'pushAction', 'action': json.dumps(action)}
+
+        return self.sendURLAsPost(data)
+
+
+    #-------------------------
+    # dispatch actions
+    #-------------------------
+
+    #----------------------------------------------------------------
+    # action on showing fits, tables, XYPlot and adding extension
+    #----------------------------------------------------------------
+
     def showFits(self, fileOnServer=None, plotId=None, additionalParams=None):
```

Comment:
Looking at [this Pythonic style guide](http://docs.python-guide.org/en/latest/writing/style/) under Function Arguments, it is more Pythonic to pass additional keyword arguments as `**kwargs` which will be a dictionary of additional named parameters, rather than to have the user construct a dictionary and pass it in.